intake_esm/core.py

```diff
@@ -157,11 +174,18 @@ def serialize(self, name, directory=None):
             json_file_name = directory / json_file_name
 
         collection_data = self._col_data.copy()
-        collection_data['catalog_file'] = csv_file_name.as_posix()
+        collection_data = _clear_old_catalog(collection_data)
         collection_data['id'] = name
 
-        print(f'Writing csv catalog to: {csv_file_name}')
-        self.df.to_csv(csv_file_name, compression='gzip', index=False)
+        if catalog_type == 'file':
+            collection_data['catalog_file'] = csv_file_name.as_posix()
+            print(f'Writing csv catalog to: {csv_file_name}')
+            self.df.to_csv(csv_file_name, compression='gzip', index=False)
+        else:
+            catalog_length = len(self.df)
+            print(f'Writing catalog with {catalog_length} entries into: {json_file_name}')
```

Comment:
Should we be reporting the `catalog_length` regardless of the `catalog_type` specified by the user?

Reply:
Good idea.